### PR TITLE
Add Conv2DTranspose layer

### PR DESCRIPTION
**Pull request by caisq**
Accompanying changes:
* Flesh out the dataFormat logic in the backend function "biasAdd".

Fixes: https://github.com/tensorflow/tfjs/issues/53

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/109)

### CODE REVIEW

**Comment by caisq** [comment]
The Travis test failure is a known flake:
https://travis-ci.org/tensorflow/tfjs-layers/builds/361814297?utm_source=github_status&utm_medium=notification

**Comment by nsthorat** [comment]
FYI to restart build:
image

**Comment by nsthorat** [comment]
I restarted it for you.

**Comment by nsthorat** [comment]
:lgtm_strong:

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved.

---

*[src/backend/tfjs_backend.ts, line 1274 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/109#-L9CRQgK4oWkhfz-MvZS:-L9CRQgK4oWkhfz-MvZT:b-t5zjr1) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/e9e418dfbaccdc84cff1bb53a6e84b06d66b53cb/src/backend/tfjs_backend.ts#L1274)):*
> ```XML
>   const biasShape = bias.shape;
> 
>   let y: Tensor;
> ```

you should wrap this in a tidy to get rid of reshapes as soon as possible

---

*[src/layers/convolutional.ts, line 460 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/109#-L9CSVNlDcUdphU0ZvRH:-L9CSVNlDcUdphU0ZvRI:bjssccg) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/e9e418dfbaccdc84cff1bb53a6e84b06d66b53cb/src/layers/convolutional.ts#L460)):*
> ```XML
>     if (this.bias != null) {
>       outputs =
>           K.biasAdd(outputs, this.bias.read(), this.dataFormat) as Tensor4D;
> ```

you're leaking tensors here, you should be wrapping this whole thing in a tidy (to get rid of intermediate outputs)

---

*[src/layers/convolutional_test.ts, line 205 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/109#-L9CS2tCAXLW51c308T6:-L9CS2tCAXLW51c308T7:b-ec8d59) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/e9e418dfbaccdc84cff1bb53a6e84b06d66b53cb/src/layers/convolutional_test.ts#L205)):*
> ```XML
>             const layer =
>                 new Conv2DTranspose({filters, kernelSize, padding, strides});
>             const y = layer.apply(x) as SymbolicTensor;
> ```

in these tests, it would be good to call `tf.memory()` to make sure you're not leaking after layer operations

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/109#-:-L9CSawd3dgn9qj-mv5G:b-jbsf8z)*

**Comment by nsthorat** [comment]
You definitely want to clean up memory in as tight a scope as possible.

We have an "@operation" decorator which automatically cleans up memory for a function which you could use, check this out: https://github.com/tensorflow/tfjs-core/blob/master/src/ops/batchnorm.ts#L38
